src/System.IO.Pipelines/src/System/IO/Pipelines/StreamPipeReader.cs

```diff
@@ -223,7 +223,7 @@ public override async ValueTask<ReadResult> ReadAsync(CancellationToken cancella
 
```

Comment:
Unrelated, but I see it above: what does the Cancel method called above do?  If it doesn't need ExecutionContext flowed to it, you could use UnsafeRegister rather than Register.  Otherwise it's fine.

Reply:
Done